### PR TITLE
update maintainers link in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes: https://github.com/kubernetes/test-infra/issues/3609

also re-formats the file a bit to not have very long lines, markdown will combine these into paragraphs anyhow

and updates the merge process description

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
+Make a [pull request](https://help.github.com/articles/about-pull-requests/) (hereafter PR). 
+In the PR body, feel free to add an area label if appropriate by saying `/area <AREA>`. 
+The list of labels is [here](https://github.com/kubernetes/test-infra/labels). 
+Also feel free to suggest a reviewer with `/assign @theirname`.
 
-Once your reviewer is happy, they will say `/lgtm`. If you do not have write access to this repo, they will merge for you. Otherwise, click the big green merge button when ready to merge. If neither you nor your reviewer has write access, please `/assign` someone from [kubernetes/test-infra-maintainers](https://github.com/orgs/kubernetes/teams/test-infra-maintainers/members) and ask them to merge for you.
+Once your reviewer is happy, they will say `/lgtm`. 
+If you do not have write access to this repo, they will merge for you. 
+Otherwise, click the big green merge button when ready to merge.
```
Comment (cblecker) [inline CONTRIBUTING.md]: Isn't this incorrect? Tide does this.

Comment (BenTheElder) [reply CONTRIBUTING.md]: Yes, fixing

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,8 +5,11 @@ In the PR body, feel free to add an area label if appropriate by saying `/area <
...
-If neither you nor your reviewer has write access, please `/assign` someone from 
-[OWNERS](https://github.com/kubernetes/test-infra/blob/master/OWNERS) and ask them to merge for you.
+Once your reviewer is happy, they will say `/lgtm` which will apply the 
+`lgtm` label, and will apply the `approved` label if they are an 
+[owner](https://github.com/kubernetes/test-infra/blob/master/OWNERS).
+The `approved` label will also automatically be applied to PRs opened by an 
+OWNER. If neither you nor your reviewer is an owner, please `/assign` someone
+ who is.
+Your PR will be automatically merged once it has the the `lgtm` and `approved` 
+labels, does not have any `hold/*` labels, and all tests are passing.
```
Comment (cblecker) [inline CONTRIBUTING.md]: `hold/*` should be `do-not-merge/*`

Comment (BenTheElder) [reply CONTRIBUTING.md]: oops, done.

**Pull request merged**